File: src/main/java/ninja/hudy/infosmog/repository/MeasurementTypeRepository.java
package ninja.hudy.infosmog.repository;

import ninja.hudy.infosmog.model.MeasurementType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface MeasurementTypeRepository extends JpaRepository<MeasurementType, Long> {

    Optional<MeasurementType> findByName(String name);
}

File: src/main/kotlin/pl/org/smok/infosmog/repository/MeasurementTypeRepository.kt
package pl.org.smok.infosmog.repository

import pl.org.smok.infosmog.model.MeasurementType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

import java.util.Optional

@Repository
interface MeasurementTypeRepository : JpaRepository<MeasurementType, Long> {

    fun findByName(name: String): Optional<MeasurementType>
}
